AustinHarris.JsonRpcTestN/Test.cs

```diff
@@ -1879,6 +1879,13 @@ public void TestNestedReturnType()
             Assert.AreEqual(expected, result.Result);
         }
 
+        [Test]
+        public void InstantiatingAServiceMoreThanOnceDoesNotThrow()
+        {
+            Assert.DoesNotThrow(()=>new CalculatorService());
+            Assert.DoesNotThrow(() => new CalculatorService());
```

Comment:
Glad you found the workarounds useful :)

I think you are right. Having an exception makes it cleaner and less hidden when something is not acceptable. What I would like to suggest is to make the error message clearer than a generic "An item with the same key ...". 
Have a look at this commit ["Make error message clearer when registering methods that have the same name"](https://github.com/alexbclarke/JSON-RPC.NET/commit/7c98955975d5a210a2d5c375c69fc2ef4db5e6eb)

I have also fixed a few tests failing simply because the expected result string was slightly miss-formatted: [Fix tests failing due to miss-formatted expected results.](https://github.com/alexbclarke/JSON-RPC.NET/commit/94b8aa1372063283f36c415dde23ef9b881f69a6)